In [ ]:
module2_lab.ipynb


In [1]:
name = "Anna"
hourly_wage = 20.0
hours = 32
weekly_pay = hourly_wage * hours
print(f"{name} earns {weekly_pay} per week.")

Anna earns 640.0 per week.


In [2]:
temps = [61, 65, 72, 58, 70, 74, 68]

warm_days = 0
for temp in temps:
    if temp > 65:
        warm_days = warm_days + 1

print(warm_days)

4


In [6]:
def to_celsius(f):
    return (f - 32) * 5/9

print(round(to_celsius(temps[0]), 1))

16.1


In [4]:
movie = {
    "title": "Titanic",
    "year": 1997,
    "rating": 7.9
}

for key, value in movie.items():
    print(key, ":", value)

title : Titanic
year : 1997
rating : 7.9


In [8]:
import pandas as pd

df = pd.read_csv("chickwts.csv")
df.head()

,Unnamed: 0,weight,feed
0,1,179,horsebean
1,2,160,horsebean
2,3,136,horsebean
3,4,227,horsebean
4,5,217,horsebean


In [7]:
%pip install pandas

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.0/10.0 MB 7.8 MB/s  0:00:01 7.5 MB/s eta 0:00:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.4/5.4 MB 9.4 MB/s  0:00:00m 9.9 MB/s eta 0:00:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2/2 [pandas]━━━━ 1/2 [pandas]
Note: you may need to restart the kernel to use updated packages.


In [9]:
import pandas as pd
df = pd.read_csv("chickwts.csv")
df.head()

,Unnamed: 0,weight,feed
0,1,179,horsebean
1,2,160,horsebean
2,3,136,horsebean
3,4,227,horsebean
4,5,217,horsebean


In [10]:
print(df.shape)
df.describe()

(71, 3)


,Unnamed: 0,weight
count,71.000000,71.000000
mean,36.000000,261.309859
std,20.639767,78.073700
min,1.000000,108.000000
25%,18.500000,204.500000
50%,36.000000,258.000000
75%,53.500000,323.500000
max,71.000000,423.000000


The typical chick weight is around 258–261. The weights vary quite a lot, from 108 to 423, so there is a wide spread in the data.